deployment/arc-k8s-extension-release-v2/ServiceGroupRoot/Scripts/arcExtensionReleaseCanaryPreview.sh

```diff
@@ -0,0 +1,241 @@
+#!/bin/bash
+# Register azuremonitor-containers extension with Arc Registration API
+export HELM_EXPERIMENTAL_OCI=1
+
+REGISTER_REGIONS_CANARY=$REGISTER_REGIONS_CANARY
+RELEASE_TRAINS_PREVIEW_PATH=${RELEASE_TRAINS_PREVIEW_PATH}
+RELEASE_TRAINS_STABLE_PATH=${RELEASE_TRAINS_STABLE_PATH}
+REGISTER_REGIONS_BATCH=($REGISTER_REGIONS_BATCH)
+IS_CUSTOMER_HIDDEN=$IS_CUSTOMER_HIDDEN
+CHART_VERSION=${CHART_VERSION}
+
+PACKAGE_CONFIG_NAME="${PACKAGE_CONFIG_NAME:-microsoft.azuremonitor.containers-pkg022022}"
+API_VERSION="${API_VERSION:-2021-05-01}"
+METHOD="${METHOD:-put}"
+REGISTRY_PATH_CANARY_PREVIEW="https://mcr.microsoft.com/azuremonitor/containerinsights/canary/preview/azuremonitor-containers"
+REGISTRY_PATH_CANARY_STABLE="https://mcr.microsoft.com/azuremonitor/containerinsights/canary/stable/azuremonitor-containers"
+REGISTRY_PATH_PROD_STABLE="https://mcr.microsoft.com/azuremonitor/containerinsights/prod1/stable/azuremonitor-containers"
+
+if [ -z "$REGISTER_REGIONS_CANARY" ]; then
+    echo "-e error release region must be provided "
+    exit 1
+fi
+if [ -z "$IS_CUSTOMER_HIDDEN" ]; then
+    echo "-e error is_customer_hidden must be provided "
+    exit 1
+fi
+if [ -z "$CHART_VERSION" ]; then
+    echo "-e error chart version must be provided "
+    exit 1
+fi
+
+MCR_NAME_PATH="mcr.microsoft.com/azuremonitor/containerinsights/canary/preview/azuremonitor-containers"
```

Comment:
i see chart verified only against canary preview repo path, this should be checked against release stage, for example if its canary check against canary repo path and if its stable release stage, then check again release stage path, right?

Reply:
updated